WooCommerce/Classes/ViewRelated/Dashboard/Free Trial/FreeTrialBannerPresenter.swift

```diff
@@ -0,0 +1,127 @@
+import Foundation
+import Combine
+import UIKit
+
+/// Presents or hides the free trial banner at the bottom of the screen.
+/// Internally uses the `storePlanSynchronizer` to know when to present or hide the banner.
+///
+final class FreeTrialBannerPresenter {
+    /// View controller used to present any action needed by the free trial banner.
+    ///
+    private weak var viewController: UIViewController?
+
+    /// View that will contain the banner.
+    ///
+    private weak var containerView: UIView?
+
+    /// Current site ID. Needed to present the upgrades web view.
+    private let siteID: Int64
+
+    /// Closure invoked when the banner is added or removed.
+    ///
+    private var onLayoutUpdated: (_ containerView: UIView, _ bannerHeight: CGFloat) -> Void
```

Comment:
Does this closure need to return the `containerView`? It doesn't look like it's being used anywhere.

Reply:
Good catch, I ended up not needing it! https://github.com/woocommerce/woocommerce-ios/pull/9521/commits/e90f2c5aa33e187423876f4ede86a48363556a5a